### PR TITLE
COV-14 Add groovy and spock tests

### PR DESCRIPTION
**Pull request by emiliyank**

### CODE REVIEW

File: src/test/groovy/CountryFactoryTest.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.covid.api.factory.CountryFactory
+import spock.lang.Specification
+
+import java.time.LocalDateTime
+
+class CountryFactoryTest extends Specification {
+
+    def factory = new CountryFactory()
+
+    def "assembleCountries() should return a list of countries with correct data"() {
```
Comment (cupakob) [inline src/test/groovy/CountryFactoryTest.groovy]: `assembleCountries()` is not needed, in the when block i should have all this information. `should` is a good start word, i do it in the same way ;)

File: src/test/groovy/CountryFactoryTest.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        when:
+        def summary = new CovidSummaryResponse()
+        summary.setId("id")
+        summary.setGlobal(new GlobalData(1, 2, 3, 4, 5, 6, "2023-01-26T00:00:00Z"))
+        summary.setCountries([new CountryData("id2", "US", "US", "united-states", 7, 8, 9, 10, 11, 12, "2023-01-26T00:00:00Z")])
+        summary.setDate("2023-01-26T00:00:00Z")
+        def countries = factory.assembleCountries(summary)
```
Comment (cupakob) [inline src/test/groovy/CountryFactoryTest.groovy]: try to use `given` and put here only the `factory.assembleCountries(summary)`

File: src/test/groovy/CovidExceptionHandlerTest.groovy [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    def "resourceNotFound() should return NOT_FOUND status and correct error message"() {
+        when:
+        def ex = new ResourceNotFoundException("Country not found")
+        def request = Mock(WebRequest)
+        request.getDescription(_) >> "Country not found"
+        def result = exceptionHandler.resourceNotFound(ex, request)
+
+        then:
+        result.statusCode.value() == HttpStatus.NOT_FOUND.value()
```
Comment (cupakob) [inline src/test/groovy/CovidExceptionHandlerTest.groovy]: 1. `HttpStatus.NOT_FOUND.value()` can be defined in the given block as variable (e.g. expectedStatusCode), same for `Country not found` and `Invalid country code`
2. (optional, no needs to be changed) In Golang the expected value is put on first place, the result value on second, this is for me personally easier to read (e.g. i expect ABC, but i got DEF)

File: src/test/groovy/CountryFactoryTest.groovy [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        then:
```
Comment (cupakob) [inline src/test/groovy/CountryFactoryTest.groovy]: please check this link:
https://spockframework.org/spock/docs/1.0/data_driven_testing.html

1. try to use data table or data pipes
2. try to check the number of the calls on a mock object (e.g. WebRequest in the next test)

File: src/test/groovy/CovidExceptionHandlerTest.groovy [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        def ex = new ValidationCountryCodeException("Invalid country code")
+        def request = Mock(WebRequest)
+        request.getDescription(_) >> "Invalid country code"
+        def result = exceptionHandler.validationCountryCode(ex, request)
+
+        then:
+        result.statusCode.value() == HttpStatus.BAD_REQUEST.value()
+        result.body.statusCode == HttpStatus.BAD_REQUEST.value()
+        result.body.message == "Invalid country code"
+        result.body.path == "Invalid country code"
```
Comment (cupakob) [inline src/test/groovy/CovidExceptionHandlerTest.groovy]: why is the path the same like the message?

Comment (emiliyank) [reply src/test/groovy/CovidExceptionHandlerTest.groovy]: Seems that there was an issue with CovidExceptionHandler class. So I replaced path(`request.getDescription(false)`) with stacktrace(`ex.getStackTrace()`).

**Pull request merged**